src/libsystemd/sd-id128/sd-id128.c

```diff
@@ -130,22 +133,137 @@ _public_ int sd_id128_get_boot(sd_id128_t *ret) {
         return 0;
 }
 
+static int get_invocation_from_keyring(sd_id128_t *ret) {
+
+        _cleanup_free_ char *description = NULL;
+        char *d, *p, *g, *u, *e;
+        unsigned long perms;
+        key_serial_t key;
+        size_t sz = 256;
+        uid_t uid;
+        gid_t gid;
+        int r, c;
+
+#define MAX_PERMS ((unsigned long) (KEY_POS_VIEW|KEY_POS_READ|KEY_POS_SEARCH| \
+                                    KEY_USR_VIEW|KEY_USR_READ|KEY_USR_SEARCH))
+
+        assert(ret);
+
+        key = request_key("user", "invocation_id", NULL, 0);
```

Comment:
According to the kernel docs, this pins down the key (https://github.com/torvalds/linux/blob/4693fc734d675c5518ea9bd4c9623db45bc37402/Documentation/security/keys-request-key.txt#L60). So no race here ;)

OTOH, it means that we're keeping the key pinned. I guess that's OK.